plasmapy/diagnostics/thomson.py

```diff
@@ -5,25 +5,38 @@
 
 __all__ = [
     "spectral_density",
+    "spectral_density_model",
 ]
 
 import astropy.constants as const
 import astropy.units as u
 import numpy as np
+import warnings
 
+from collections import namedtuple
+from lmfit import Model
 from typing import List, Tuple, Union
 
 from plasmapy.formulary.dielectric import permittivity_1D_Maxwellian
 from plasmapy.formulary.parameters import plasma_frequency, thermal_speed
 from plasmapy.particles import Particle
 from plasmapy.utils.decorators import validate_quantities
 
+# Define some constants
+C = const.c.si  # speed of light
+
+
 # TODO: interface for inputting a multi-species configuration could be
 # simplified using the plasmapy.classes.plasma_base class if that class
 # included ion and electron drift velocities and information about the ion
 # atomic species.
 
 
+# TODO: If we can make this object pickle-able then we can set
+# workers=-1 as a kw to differential_evolution to parallelize execution for fitting!
+# The probem is a lambda function used in the Particle class...
+
+
 @validate_quantities(
```

Comment:
We've recently add a [`can_be_zero` check to `validate_quantities`](https://docs.plasmapy.org/en/latest/api/plasmapy.utils.decorators.validate_quantities.html#validate-quantities) that would be a good option for `wavelength`, `probe_wavelength` and `n`.

Reply:
Sounds good - done.